tgui/packages/tgui/interfaces/ResearchTerminal.tsx

```diff
@@ -0,0 +1,588 @@
+  }
+
+  const iconRef = (name: string, isNum: boolean) => sortby === name
+    ? sortdir === 'asc'
+      ? isNum ? "arrow-down-1-9" : "arrow-down-a-z" // small to big
+      : isNum ? "arrow-down-9-1" : "arrow-down-z-a" // big to small
+    : "space";
+
+  const sortColClick = (name: string) => {
+    if (sortby === name) {
+      setSortdir(sortdir === 'asc' ? 'desc' : 'asc');
+    } else {
+      setSortBy(name);
+    }
+  };
+
+
+  return (
+    <Table className={props.className}>
+      <TableRow>
+        <TableCell textAlign="center">
+          <Button
+            icon={iconRef('time', true)}
+            onClick={() => sortColClick('time')}
+          >
+            {props.timeLabel}
+          </Button>
+        </TableCell>
+        <TableCell textAlign="center">
+          <span>
+            Type
+          </span>
+        </TableCell>
+        <TableCell textAlign="center">
+          <Button icon={iconRef('name', false)} onClick={() => sortColClick('name')}>
+            Compound
+          </Button>
+        </TableCell>
+        <TableCell textAlign="center">
+          <span>Actions</span>
+        </TableCell>
+      </TableRow>
+      {Array.from(outputDocs.values())
+        .sort((a, b) => {
+          if (sortby === "time") {
+            if (sortdir === "asc") {
+              return a.type.time < b.type.time ? -1 : 1;
+            } else {
+              return a.type.time > b.type.time ? -1 : 1;
+            }
+          } else {
+            if (sortdir === "asc") {
+              return a.type.document.localeCompare(b.type.document);
+            } else {
+              return b.type.document.localeCompare(a.type.document);
+            }
+
+          }
+        })
+        .map(x => <CompoundRecord compound={x} key={x.id} canPrint={props.canPrint} />)}
+    </Table>);
+};
+
+
+const PhotocopierMissing = () => {
+  return (<span>
+    ERROR: no linked printer found.
+          </span>);
+};
+
+const TonerEmpty = () => {
+  return (<span>
+    ERROR: Printer toner is empty.
+          </span>);
+};
+
+const ImproveClearanceConfirmation = (props, context) => {
+  const { data, act } = useBackend<TerminalProps>(context);
+  const [isConfirm, setConfirm] = useLocalState<string|undefined>(context, 'purchase_confirmation', undefined);
+  if (isConfirm === undefined || isConfirm !== "broker_clearance") {
+    return null;
+  }
+  return (
+    <Stack vertical>
+        <Stack.Item>
+          <ConfirmationDialogue
+            onConfirm={() => {
+              act("broker_clearance");
+              setConfirm(undefined);
+            }}
+            onCancel={() => setConfirm(undefined)}
+          >
+            <span>
+              The CL can swipe their ID card on the console to increase
+              clearance for free, given enough DEFCON. Are you sure you want to
+              spend <u>{data.broker_cost}</u> research credits to increase the
+              clearance immediately?
+            </span>
```

Comment:
```suggestion
            <span>
              Are you sure you want to spend <u>{data.broker_cost}</u> research 
              credits to increase the clearance immediately?
            </span>
```
the ship CL doesn't have this ability regardless of intel level - and is a surv CL thing. this message could be made to reference the ability of the survivor CL to increase clearance, but i'm not sure how you do that neatly

Reply:
I just copied the existing text, probably just best to ignore survivor cl info for rp reasons